liangBarskyAlgo/liangBarskyAlgo.c

```diff
@@ -0,0 +1,97 @@
+// Header files included
+#include<stdio.h>
+#include<conio.h>
+#include<graphics.h>
+
+void main()
+{
+    int gd = DETECT, gm;
+    /** x1, x2, y1 & y2 are line coordinates
+        xmax, xmin, ymax & ymin are window's lower and upper corner coordinates
+        xx1, yy1, xx2 & yy2 are the new coordinates of clipped line
+    **/
+    int x1, y1, x2, y2, xmax, xmin, ymax, ymin, xx1, yy1, xx2, yy2, dx, dy, i;
+    int p[4], q[4];
+    float t1, t2, t[4];
+
+    initgraph (& gd, & gm, "C:\\TC\\BGI");
+    // Enter the coordinates for window
+    printf("Enter the lower co-ordinates of window");
+    scanf("%d", &xmin, &ymin);
+    printf("Enter the upper co-ordinates of window");
+    scanf("%d", &xmax, &ymax);
+
+    // Graphics function to draw rectangle/clipping window
+    setcolor(RED);
+    rectangle(xmin, ymin, xmax, ymax);
+
+    // Enter the coordinates of line
+    printf("Enter x1:");
+    scanf("%d", &x1);
+    printf("Enter y1:");
+    scanf("%d", &y1);
+    printf("Enter x2:");
+    scanf("%d", &x2);
+    printf("Enter y2:");
+    scanf("%d", &y2);
+
+    // Graphics Function to draw line
+    line(x1, y1, x2, y2);
+
+    dx = x2 - x1;
+    dy = y2 - y1;
+    p[0] = - dx;
+    p[1] = dx;
+    p[2] = - dy;
+    p[3] = dy;
+    q[0] = x1 - xmin;
+    q[1] = xmax - x1;
+    q[2] = y1 - ymin;
+    q[3] = ymax - y1;
+
+    for (i=0; i < 4; i++) {
+
+        if (p[i] != 0) {
+            t[i] = (float) q[i] / p[i];
+        }
+
+        else {
+            if (p[i] == 0 && q[i] < 0)
+                printf ("line completely outside the window");
```

Comment:
Since you have not added space after ``printf`` in the statements before this one, I'd suggest you to remove it from this statement onwards so that there is unification throughout the code.